lib/init/grass.py

```diff
@@ -1854,8 +1857,10 @@ def sh_like_startup(location, location_name, grass_env_file, sh):
     f = open(shell_rc_file, 'w')
 
     if sh == 'zsh':
-        if not os.getenv('SAVEHIST'):
-            os.environ['SAVEHIST'] = os.getenv('HISTSIZE')
+        # zsh does not have an unlimited history setting, so 1e8 is set as a proxy
+        os.environ['SAVEHIST'] = "100000000"
+        os.environ['HISTSIZE'] = os.getenv('SAVEHIST')
```

Comment:
It seems to me now that this history handling should be together with the `if sh == "bash":` history handling above (e.g., in elif).